docs/source/advanced_config.rst

```diff
@@ -6,17 +6,64 @@ Advanced Config
 
 Custom Steps
 ============
-coming soon
+If you need a custom build step, you can create a subclass of the :class:`~fab.steps.Step` class.
 
-Metrics
-=======
-coming soon
+Fab includes some examples of a custom step. A simple example was created for building JULES.
+The :class:`~fab.steps.root_inc_files.RootIncFiles` step copies all `.inc` files in the source tree
+into the root of the source tree, to make subsequent preprocessing flags easier to configure.
+
+That was a simple example that didn't need to interact with the :term:`Artefact Store`.
+If you need to insert a custom step between two Fab steps, you may need your custom step to create
+an :term:`Artefact Collection`, and to configure the subsequent step to use your new collection.
+Many Fab steps have an optional `source` argument which overrides the default :term:`Artefacts Getter`.
```

Comment:
:+1: How's this instead?

> That was a simple example that didn't need to interact with the `Artefact Store`.
> Sometimes, inserting a custom step means inserting a new `Artefact Collection` into the flow of data between
> steps. We can tell a subsequent step to read our new artefacts, instead of using it's default `Artefacts Getter`.
> We do this using the `source` argument, which most Fab steps accept.
> 

I wonder if the general terminology could do with a once-over, too.